File: advance/1040.cpp
#include <iostream>
#include <string.h>
#include <stdio.h>


using namespace std;

char s1[1001];
char s2[1001];

int main() {
    char c;
    int k = 0;
    while((c = getchar()) != '\n') {
        s1[k++] = c;
    }
    int length = strlen(s1);
    for(int i = 0; i < length; ++i) {
        s2[i] = s1[length - i - 1];
    }
    int maxl = 0;
    int i = 0;
    while (i < length) {
        int lasti = i;
        int count = 0;
        int j = 0;
        while(j < length) {
            int ccount = 0;
            int lastj = j;
            while (i < length && j < length && s1[i] == s2[j]) {
                i ++;
                j ++;
                ccount += 1;
            }
            if(ccount > count) {
                count = ccount;
            }
            j ++;
            i = lasti;
        }
        if(count > maxl) {
            maxl = count;
        }
        i = lasti + 1;
    }
    cout << maxl << endl;
}

File: advance/1040.py
a = input()
b = list(a)
b.reverse()
# print(a)
# print("".join(b))

maxl = 0
i = 0
while i < len(a):
    lasti = i
    count = 0
    j = 0
    while j < len(b):
        ccount = 0
        lastj = j
        while i < len(a) and j < len(b) and a[i] == b[j]:
            i += 1
            j += 1
            ccount += 1
        if ccount > count:
            count = ccount
        j += 1
        i = lasti
    if count > maxl:
        maxl = count
    i = lasti + 1
print(maxl)

